### DISCUSSION TITLE
Delete user

### DISCUSSION THREAD

**Discussion by tpapamichail** [drakkan/sftpgo#288] [Q&A] [answered]
When we delete a user, his folder and his data are not deleted.
Am I doing something wrong?

**Comment by drakkan** [comment, marked as answer]
Hi,

this is by design, deleting data can be dangerous (for example you delete an user by mistake) and depending on the amount of data it could also take a long time.

However you can use a custom [hook](https://github.com/drakkan/sftpgo/blob/master/docs/custom-actions.md) to customize things

"""
The actions struct inside the "data_provider" configuration section allows you to configure actions on user add, update, delete.
"""
